guides/common/modules/con_planning-project-upgrade.adoc

```diff
@@ -0,0 +1,18 @@
+[id="planning-{project-context}-upgrade"]
+= Planning {Project} upgrade
+
+Upgrading to {Project} {ProjectVersion} affects your entire {Project} infrastructure because {Project} services are shut down during the upgrade.
+The upgrade process duration varies depending on your hardware configuration, network speed, and the amount of data that is stored on the server.
+Plan carefully before proceeding.
+
+* Read the {ProjectName} {ProjectVersion} {ReleaseNotesDocURL}[Release Notes].
+* Consider whether any of your scripts need updating.
+Some {Project} API endpoints, Hammer CLI commands, and modules from the {Project} Ansible Collection can differ between versions of {Project}.
+ifdef::satellite[]
+For information about changes in these tools, see the {ProjectName} {ProjectVersion} {ReleaseNotesDocURL}[Release Notes].
+endif::[]
```

Comment:
On line 8 we also point to the release notes and this feels like a repetition. I wonder if there's a way to combine them.

Reply:
I wondered myself and couldn't figure out a way. The first reference is generic. This second one talks about a specific piece of information that's included in the release notes, and only downstream. That's why I kept them separate. If anyone has a better suggestion, I'll consider it. For now, no action.

Reply:
Perhaps move away from bullet points into a paragraph?

Read the {ProjectName} {ProjectVersion} {ReleaseNotesDocURL}[Release Notes].
This helps you plan your upgrade by pointing out noteworthy changes.
For example, {Project} API endpoints, Hammer CLI commands, and modules from the {Project} Ansible Collection can differ between versions of {Project}.
Any custom integrations, such as scripts may need to be updated.

The above IMHO doesn't need any conditionals and explains how to view the release notes. I'm sure you can improve wording.

Reply:
I don't think dropping the conditional works because upstream notes actually don't list the changes, we only publish the changelists downstream (https://content-preview.docs.redhat.com/en/documentation/red_hat_satellite/6.17/html-single/release_notes/index#changes-in-satellite-installer-parameters-hammer-cli-api-and-ansible_satellite). That reference is something I can only include for satellite.

Also, I like how both actions have their own bullet point: 1. Read the notes. 2. Think about your integrations. Both are actions users should carefully consider so I believe both deserve their own emphasis.

I agree that including the RN link twice is not the most elegant solution but it's functional. And it's also only for downstream satellite so any potential impact is limited.